cfile/cfile.cpp

```diff
@@ -69,6 +79,112 @@ cfile_error cfe;
 // The message for unexpected end of file
 const char *eof_error = "Unexpected end of file";
 
+// This function must be called before you use anything else from this module.
+void cf_Init(std::vector<std::filesystem::path> initial_base_directories) {
+  Base_directories = initial_base_directories;
+}
+
+#ifdef WIN32
+std::filesystem::path cf_LocatePathCaseInsensitiveHelper(const std::filesystem::path &relative_path, const std::filesystem::path &starting_dir) {
+  return starting_dir / relative_path;
+}
+#else
+std::filesystem::path cf_LocatePathCaseInsensitiveHelper(const std::filesystem::path &relative_path, const std::filesystem::path &starting_dir) {
+  // Dumb check, maybe there already all ok?
+  if (exists((starting_dir / relative_path))) {
+    return starting_dir / relative_path;
+  }
+
+  std::filesystem::path result, search_path, search_file;
+
+  search_path = starting_dir / relative_path.parent_path();
+  search_file = relative_path.filename();
+
+  // If directory does not exist, nothing to search.
+  if (!std::filesystem::is_directory(search_path) || search_file.empty()) {
+    return {};
+  }
+
+
+  // Search component in search_path
+  auto const &it = std::filesystem::directory_iterator(search_path);
+
+  auto found = std::find_if(it, end(it), [&search_file, &search_path, &result](const auto& dir_entry) {
+    return stricmp(dir_entry.path().filename().u8string().c_str(), search_file.u8string().c_str()) == 0;
+  });
+
+  if (found != end(it)) {
+    // Match, append to result
+    result = found->path();
+    search_path = result;
+  } else {
+    // Component not found, mission failed
+    return {};
+  }
+
+  return result;
+}
+#endif
+
+std::vector<std::filesystem::path> cf_LocatePathMultiplePathsHelper(std::filesystem::path relative_path, bool stop_after_first_result) {
+  ASSERT(("realative_path should be a relative path.", relative_path.is_relative()));
+  std::vector<std::filesystem::path> return_value = { };
+  for (auto base_directory : Base_directories) {
```

Comment:
In order to switch to `.push_back()` like you asked in another review comment, I had to change that for loop so that it iterates over `Base_directories` backwards. What do you think of that for loop now?